aws/resource_aws_media_connect_flow.go

```diff
@@ -0,0 +1,526 @@
+		Refresh: refreshMediaConnectFlowStatus(conn, d.Id()),
+	}
+	_, err := stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for updating Media Connect Flow(%s) Source: %s", d.Id(), err)
+	}
+
+	return resourceAWSMediaConnectFlowRead(d, meta)
+}
+
+func resourceAWSMediaConnectFlowDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediaconnectconn
+
+	deleOpts := &mediaconnect.DeleteFlowInput{
+		FlowArn: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Media Connect Flow delete configuration: %#v", deleOpts)
+	if _, err := conn.DeleteFlow(deleOpts); err != nil {
+		if isAWSErr(err, mediaconnect.ErrCodeNotFoundException, "") {
+			return nil
+		}
+		return fmt.Errorf("Error deleting Media Connect Flow %s: %s", d.Id(), err)
+	}
+
+	return waitForDeleteMediaConnectFlow(conn, d.Id(), d.Timeout(schema.TimeoutDelete))
+}
+
+func updateAWSMediaConnectFlowSource(conn *mediaconnect.MediaConnect, flowArn string, configList []interface{}) error {
+	config := configList[0].(map[string]interface{})
+	updateSourceOpts := &mediaconnect.UpdateFlowSourceInput{
+		FlowArn:   aws.String(flowArn),
+		SourceArn: aws.String(config["arn"].(string)),
+	}
+	if v, ok := config["decryption"]; ok {
+		updateSourceOpts.Decryption = expandAWSMediaConnectFlowSourceUpdateEncryption(v.([]interface{}))
+	}
+	if v, ok := config["entitlement_arn"]; ok && v.(string) != "" {
+		updateSourceOpts.EntitlementArn = aws.String(v.(string))
+	} else {
+		if v, ok := config["description"]; ok {
+			updateSourceOpts.Description = aws.String(v.(string))
+		}
+		if v, ok := config["ingest_port"]; ok && v.(int) != 0 {
+			updateSourceOpts.IngestPort = aws.Int64(int64(v.(int)))
+		}
+		if v, ok := config["max_bitrate"]; ok && v.(int) != 0 {
+			updateSourceOpts.MaxBitrate = aws.Int64(int64(v.(int)))
+		}
+		if v, ok := config["max_latency"]; ok && v.(int) != 0 {
+			updateSourceOpts.MaxLatency = aws.Int64(int64(v.(int)))
+		}
+		if v, ok := config["protocol"]; ok && v.(string) != "" {
+			updateSourceOpts.Protocol = aws.String(v.(string))
+		}
+		if v, ok := config["stream_id"]; ok && v.(string) != "" {
+			updateSourceOpts.StreamId = aws.String(v.(string))
+		}
+		if v, ok := config["whitelist_cidr"]; ok && v.(string) != "" {
+			updateSourceOpts.WhitelistCidr = aws.String(v.(string))
+		}
+	}
+
+	log.Printf("[DEBUG] Media Connect Flow Source update configuration: %#v", updateSourceOpts)
+	if _, err := conn.UpdateFlowSource(updateSourceOpts); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func refreshMediaConnectFlowStatus(conn *mediaconnect.MediaConnect, flowArn string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		output, err := conn.DescribeFlow(&mediaconnect.DescribeFlowInput{
+			FlowArn: aws.String(flowArn),
+		})
+		if err != nil {
+			return 42, "", err
+		}
+		flow := output.Flow
+		if flow == nil {
+			return flow, "", fmt.Errorf("Media Connect Flow (%s) missing", flowArn)
+		}
```

Comment:
To prevent panics and allow missing flow data to return successfully:

```suggestion
		if output == nil || output.Flow == nil {
			return nil, "", nil
		}
		
		flow := output.Flow
```

Reply:
fixed e6dc835